Update/quota mgmt transactions updates (#4639)
getsentry/sentry-docs: src/docs/product/accounts/quotas/manage-event-stream-guide.mdx

**PHP** - The `error_types` configuration option allows you to set the error types you want Sentry to monitor. Learn more in PHP [Basic Options](/platforms/php/configuration/options/#common-options).

**Ruby** - The `excluded_exceptions` configuration option allows you to set the exception types you wish to suppress. Learn more in the [Ruby configuration docs](/platforms/ruby/configuration/options/#optional-settings)

## 3. Inbound Data Filters

While SDK configuration requires changes to your source code and depends on your next deployment, server-side filters can be easily configured per project in the "Data Filters" section of **[Project] > Settings > Inbound Filters**. Learn more in [Inbound Filters](/product/accounts/quotas/#inbound-data-filters).

Once applied, you can track the filtered events (numbers and cause) using the graph provided at the top of the "Inbound Filters" page.

![Built-in Inbound Filters](manage-event-stream-03.png)

## 4. Event Grouping

> Proper event grouping is essential to maintaining a meaningful set of issues and to reduce redundant notifications. Sentry groups similar _events_ into unique _[issues](/product/issues/)_ based on their _fingerprint_. An event's fingerprint relies firstly on its _stack trace_.

Author: ```suggestion
Proper event grouping is essential to maintaining a meaningful set of issues and reducing redundant notifications. Sentry groups similar error events into unique _[issues](/product/issues/)_ based on their _fingerprint_. An event's fingerprint relies firstly on its _stack trace_.
```

With JavaScript errors, a minified source code might result in a nondeterministic stack trace that could negatively affect associated event grouping. Make sure Sentry has access to your source maps and minified artifacts. Learn more in [Source Maps](/platforms/javascript/sourcemaps/).

![JavaScript stack trace without source maps](manage-event-stream-04.png)

![JavaScript stack trace with source maps](manage-event-stream-05.png)

For more information on the fingerprint algorithm and customizing event grouping, check out our [Issue Grouping docs](/product/data-management-settings/event-grouping/).

## 5. Resolve or Delete Issues {#5-applying-workflows}

> Now that your error events have been fine-tuned to reflect real problems in your code, it's an excellent practice to react to errors as they happen. If an issue reflects a real problem in your code, resolve it; otherwise, discard it.

Author: ```suggestion
Now that your error events have been fine-tuned to reflect real problems in your code, it's an excellent practice to react to errors as they happen. If an issue reflects a real problem in your code, resolve it; otherwise, delete and discard it.
```

### Managing a Spike {#-spike-protection-was-activated----what-should-i-do}

You might receive an email notifying you that spike protection was triggered and applied to your account.

![Spike Protection Email](spike-email.jpg)

Many times, an unexpected spike is caused by a new error (or errors) introduced into your code with a new release version. We understand that errors can come in spikes and service interruptions can be challenging, so we include a one-time grace period for all paid accounts.

If an account depletes its event capacity and has never previously triggered a grace period, we continue to accept events for the next three days at a maximum rate of 10,000 events per hour. If you choose not to increase your capacity, Sentry will stop accepting events for the remainder of the current billing month. However, if you increase your capacity, events accepted during the grace period will be counted as consumed capacity.

Once you've used your grace period, we recommend using our on-demand spending caps to ensure you have time to adjust your capacity in the event of a future spike in errors.

> Also, consider doing the following:
> 

Author: ```suggestion
```

> - Set better rate limits on the DSN keys for the projects related to the spike. - If it's a specific release version that has caused the spike, add the version identifier to the project's inbound filters to avoid accepting events from that release.

Author: ```suggestion
- Set better rate limits on the DSN keys for the projects related to the spike. 
- If it's a specific release version that has caused the spike, add the version identifier to the project's inbound filters to avoid accepting events from that release.
```

Learn more about [how spike protection works](/product/accounts/quotas/#spike-protection).

## 6. Event Usage Stats {#common-workflows-for-managing-your-event-stream}

Once you've completed the steps above, you can look at your events in aggregate in the "Usage Stats" tab of **Stats**. This information will help you answer some key questions like:

- What's the breakdowndown of my incoming events?
- Which projects are consuming my quota?
- Which issues are consuming my quota?

The answers to these questions can help you figure out where you need to do further fine-tuning of your filters, limits, or SDK configuration.

### How can I see a breakdown of incoming events? {#-how-can-i-see-a-breakdown-of-incoming-events}

> The "[Usage Stats](/product/stats/#usage-stats)" tab displays details about the total number of events Sentry has received across your entire organization for up to 90 days. The page breaks down the events (by project) into three categories:

Author: ```suggestion
The [Usage Stats](/product/stats/#usage-stats) tab displays details about the total number of events Sentry has received across your entire organization for up to 90 days. The page breaks down the events (by project) into three categories:
```

---
title: Manage Your Error Quota
keywords: ["best practices"]
sidebar_order: 10
redirect_from:
  - /guides/manage-event-stream/
description: "Learn how to use the tools Sentry provides to control the type and amount of errors that are monitored."
---

> Sending all your application errors to Sentry ensures you'll be notified in real-time when errors occur in your code. However, with just the basic setup, your applications might generate many errors, and those might trigger more notifications than you want. Sentry provides tools to control the _type_ and _amount_ of errors that are monitored. Theese tools allow you to:

Author: ```suggestion
Sending all your application errors to Sentry ensures you'll be notified in real-time when errors occur in your code. However, with just the basic setup, your applications might generate many errors, and those might trigger more notifications than you want and use up your error quota. Sentry provides tools to control the _type_ and _amount_ of errors that are monitored. These tools allow you to:
```

> Sending all your application errors to Sentry ensures you'll be notified in real-time when errors occur in your code. However, with just the basic setup, your applications might generate many errors, and those might trigger more notifications than you want and use up your error quota. Sentry provides tools to control the _type_ and _amount_ of errors that are monitored. These tools allow you to:

Author: This is better!

1. Receive more actionable and meaningful error events.
2. Reserve real-time notifications for errors that actually break your code.

> 3. Manage your event quotas if you're on an quota-based plan.

Author: ```suggestion
3. Manage your error quota.
```

**Ruby** - The `excluded_exceptions` configuration option allows you to set the exception types you wish to suppress. Learn more in the [Ruby configuration docs](/platforms/ruby/configuration/options/#optional-settings)

## 3. Inbound Data Filters

While SDK configuration requires changes to your source code and depends on your next deployment, server-side filters can be easily configured per project in the "Data Filters" section of **[Project] > Settings > Inbound Filters**. Learn more in [Inbound Filters](/product/accounts/quotas/#inbound-data-filters).

Once applied, you can track the filtered events (numbers and cause) using the graph provided at the top of the "Inbound Filters" page.

![Built-in Inbound Filters](manage-event-stream-03.png)

## 4. Event Grouping

Proper event grouping is essential to maintaining a meaningful set of issues and reducing redundant notifications. Sentry groups similar error events into unique _[issues](/product/issues/)_ based on their _fingerprint_. An event's fingerprint relies firstly on its _stack trace_.

> With JavaScript errors, a minified source code might result in a nondeterministic stack trace that could negatively affect associated event grouping. Make sure Sentry has access to your source maps and minified artifacts. Learn more in [Source Maps](/platforms/javascript/sourcemaps/).

Reviewer: ```suggestion
With JavaScript errors, minified source code might result in a nondeterministic stack trace that could negatively affect associated event grouping. To improve grouping, make sure Sentry has access to your source maps and minified artifacts. Learn more in [Source Maps](/platforms/javascript/sourcemaps/).
```

In **[Project] > Settings > Client Keys (DSN)**, click "Configure", and you can create multiple DSN keys per project and assign different (or no) rate limits to each key. This will allow you to dynamically allocate keys (with varying thresholds) depending on release, environment, and so on.

![Per DSN Key rate limits](manage-event-stream-11.png)

### Setting Useful Rate Limits {#-how-to-set-proper-rate-limits}

A good way to set a project rate limit is by figuring out the expected event volume based on your average traffic. Let's look at an example:

![Calculating rate limits](manage-event-stream-14.png)

- Open the project DSN key configuration under **[Project] > Settings > Client Keys (DSN)** by clicking "Configure".
- In the `KEY USAGE IN THE LAST 30 DAYS` graph and look for the highest point, or the maximum daily rate. In this example, the maximum daily rate in the last month is less than 326K.
- Based on that, we can define a ceiling **daily** maximum value of approximantely 330K, which is around 13,750 events an **hour**, or  about 230 events a **minute**.

> - You can set a daily, hourly, or minute-based rate limit. We'd recommend using a minute-based rate to avoid situations where a random event spike might exhaust your daily or hourly set quota and leave you blind for a long while.

Reviewer: ```suggestion
- You can set a daily, hourly, or minute-based rate limit. We'd recommend using a minute-based rate to avoid situations where a random event spike might exhaust your daily or hourly quota and leave you blind for a long period.
```

🤷
